src/main/java/co/cask/spanner/common/SpannerConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.common;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.ReferencePluginConfig;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Spanner configuration
+ */
+public class SpannerConfig extends ReferencePluginConfig {
+  @Name("project")
+  @Description("Project ID")
+  @Macro
+  public String project;
+
+  @Name("instance")
+  @Description("Spanner Instance Id")
+  @Macro
+  public String instance;
+
+  @Name("database")
+  @Description("Database to be used")
+  @Macro
+  public String database;
+
+  @Name("serviceFilePath")
+  @Description("Service Account File Path")
```

Comment:
Need more description here.

Path on the local file system to the service account key used for access to Spanner. Does not need to be specified when running on a Dataproc cluster. When running on other clusters, the file must be present on every node in the cluster.